src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -391,12 +762,16 @@ void ScatterUpdate::execute(dnnl::stream strm) {
             break;
         }
         case ScatterUpdateMode::ScatterElementsUpdate: {
-            scatterElementsUpdate(indicesPtr, updatePtr, axis, dstPtr);
+            using namespace ov::element;
+            IF_TYPE_OF(scatter_el_update_data_type,
+                OV_PP_ET_LIST(f32, bf16, f16, i32),
+                scatter_elements_update::Caller,
+                dataPrec, indicesPrec,
+                dstMemPtr, indicesMemPtr, updateMemPtr, axis, this->m_config);
```

Comment:
@dmitry-gorokhov

Reply:
Is `IF_TYPE_OF` really good for better readability & debuggability?

Reply:
This is to make the template instantiate simple. Without it, we have to use nested switch-case for each combination of DataType and IndexType. Any other idea to avoid this swelling nested switch-case?

Reply:
refacted.